File: atoystore-client/src/types/Order.ts
export enum OrderStatus {
  Pending = 0,
  Processing = 1,
  Paid = 2,
  Shipped = 3,
  Completed = 4,
  Cancelled = 5
}

export interface OrderItem {
  product: {
    id: string;
    title: string;
    price: number;
  };
  quantity: number;
  unitPrice: number;
}

export interface PaymentInfoDto {
  method: "Cash" | "YooKassa";
  isPaid: boolean;
}

export interface ShippingInfoDto {
  address: string;
  city: string;
  country: string;
}

export interface Order {
  id: string;
  userId: string;
  customerName: string;
  customerPhone: string;
  note: string;
  createdAt: string;
  totalPrice: number;
  status: OrderStatus;
  shipping: {
    address: string;
    city: string;
    country: string;
  };
  items: OrderItem[];
  payment: PaymentInfoDto; // Добавлено для поддержки онлайн-оплаты
}

export interface OrderItemDto {
  productId: string;
  quantity: number;
  unitPrice: number;
}

export interface CreateOrderDto {
  userId: string;
  customerName: string;
  customerPhone: string;
  totalPrice: number;
  items: OrderItemDto[];
  shipping: ShippingInfoDto;
  payment: PaymentInfoDto;
  note?: string;
}

File: AToyStore.API/AToyStore.Core/Orders/Entities/Order.cs
namespace AToyStore.Core.Orders.Entities
{
    public class Order
    {
        public Guid Id { get; set; }
        public string UserId { get; set; } = string.Empty;
        public string CustomerName { get; set; } = string.Empty;
        public string CustomerPhone { get; set; } = string.Empty;
        public string Note { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public OrderStatus Status { get; set; } = OrderStatus.Pending;

        public decimal TotalPrice { get; set; }

        public List<OrderItem> Items { get; set; } = new();
        public PaymentInfo Payment { get; set; } = new();
        public ShippingInfo Shipping { get; set; } = new();
    }
}
